pkg/image/validation/kubernetes_test.go

```diff
@@ -1079,3 +1070,691 @@ func TestValidateAdditionalArtifacts(t *testing.T) {
+			assert.Len(t, failures, len(test.ExpectedFailedMessages))
+
+			var foundMessages []string
+			for _, foundValidation := range failures {
+				foundMessages = append(foundMessages, foundValidation.UserMessage)
+			}
+
+			for _, expectedMessage := range test.ExpectedFailedMessages {
+				assert.Contains(t, foundMessages, expectedMessage)
+			}
+
+		})
+	}
+}
+
+func TestValidateNodeIP(t *testing.T) {
+	tests := map[string]struct {
+		K8s                    image.Kubernetes
+		ServerConfig           map[string]any
+		ExpectedFailedMessages []string
+	}{
+		`single IPv4`: {
+			K8s: image.Kubernetes{},
+			ServerConfig: map[string]any{
+				"node-ip": "10.42.0.0",
+			},
+		},
+		`single IPv6`: {
+			K8s: image.Kubernetes{},
+			ServerConfig: map[string]any{
+				"node-ip": "fd12:3456:789a::21",
+			},
+		},
+		`multiple nodes valid`: {
+			K8s: image.Kubernetes{
+				Nodes: []image.Node{
+					{
+						Hostname: "server1",
+						Type:     image.KubernetesNodeTypeServer,
+					},
+					{
+						Hostname:    "agent",
+						Type:        image.KubernetesNodeTypeAgent,
+						Initialiser: true,
+					},
+					{
+						Hostname:    "agent2",
+						Type:        image.KubernetesNodeTypeAgent,
+						Initialiser: true,
+					},
+				},
+			},
+			ServerConfig: map[string]any{
+				"node-ip": "10.42.0.0",
+			},
+		},
+		`multiple servers invalid`: {
+			K8s: image.Kubernetes{
+				Nodes: []image.Node{
+					{
+						Hostname: "server1",
+						Type:     image.KubernetesNodeTypeServer,
+					},
+					{
+						Hostname:    "server2",
+						Type:        image.KubernetesNodeTypeServer,
+						Initialiser: true,
+					},
+				},
+			},
+			ServerConfig: map[string]any{
+				"node-ip": "10.42.0.0",
+			},
+			ExpectedFailedMessages: []string{
+				"Kubernetes server config node-ip can not be specified when there is more than one Kubernetes server node",
+			},
+		},
+		`node ip family both same invalid`: {
+			K8s: image.Kubernetes{},
+			ServerConfig: map[string]any{
+				"node-ip": "10.42.0.0,10.43.0.0",
+			},
+			ExpectedFailedMessages: []string{
+				"Kubernetes server config node-ip cannot contain addresses of the same IP address family; one must be IPv4, and the other IPv6",
+			},
+		},
+		`node ip dualstack valid`: {
+			K8s: image.Kubernetes{},
+			ServerConfig: map[string]any{
+				"node-ip": "10.42.0.0,fd12:3456:789a::21",
+			},
+		},
+		`node ip non-unicast IPv4 invalid`: {
+			K8s: image.Kubernetes{},
+			ServerConfig: map[string]any{
+				"node-ip": "127.0.0.1,fd12:3456:789a::21",
+			},
+			ExpectedFailedMessages: []string{
+				"Kubernetes server config node-ip value '127.0.0.1' must be a valid unicast address",
+			},
+		},
+		`node ip IPv6 invalid`: {
+			K8s: image.Kubernetes{},
+			ServerConfig: map[string]any{
+				"node-ip": "xxxx:3456:789a::21",
+			},
+			ExpectedFailedMessages: []string{
+				"Kubernetes server config node-ip value 'xxxx:3456:789a::21' could not be parsed",
+			},
+		},
+	}
```

Comment:
These two are confusing. The first one is dual-stack and implies we should also test a non-unicast IPv6. The second one is single-stack and implies we should also test an invalid IPv4. Are we missing those cases on purpose?